polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -644,7 +644,9 @@ ValueCategory MLIRScanner::CommonArrayToPointer(ValueCategory Scalar) {
 ValueCategory MLIRScanner::CommonArrayLookup(ValueCategory Array, Value Idx,
                                              bool IsImplicitRefResult,
                                              bool RemoveIndex) {
-  Value Val = Array.getValue(Builder);
+  Value Val = isa<MemRefType, LLVM::LLVMPointerType>(Array.getElemTy())
```

Comment:
Nit: Can we have a comment here explaining which cases are handled here?

Reply:
Better, but maybe we could make clear what kind of input code would lead to the two scenarios?